packages/cli/test/simulation/multi_fork.test.ts

```diff
@@ -105,11 +107,43 @@ await checkpointSync.jobs.el.start();
 await checkpointSync.jobs.cl.start();
 await connectNewNode(checkpointSync.nodePair, env.nodes);
 
-await env.waitForNodeSync(rangeSync.nodePair);
-await env.waitForNodeSync(checkpointSync.nodePair);
-
+await waitForNodeSync(rangeSync.nodePair, env.options.controller.signal);
+await waitForNodeSync(checkpointSync.nodePair, env.options.controller.signal);
 await rangeSync.jobs.cl.stop();
 await rangeSync.jobs.el.stop();
 await checkpointSync.jobs.cl.stop();
 await checkpointSync.jobs.el.stop();
+
+// Unknown block sync
+// ========================================================
+const unknownBlockSync = env.createNodePair({
+  id: "unknown-block-sync-node",
+  cl: {type: CLClient.Lodestar, options: {"network.allowPublishToZeroPeers": true, "sync.disableRangeSync": true}},
+  el: ELClient.Geth,
+  keysCount: 0,
+});
+await unknownBlockSync.jobs.el.start();
+await unknownBlockSync.jobs.cl.start();
+const head = await env.nodes[0].cl.api.beacon.getBlockV2("head");
+await connectNewNode(unknownBlockSync.nodePair, env.nodes);
+
+try {
+  await unknownBlockSync.nodePair.cl.api.beacon.publishBlock(head.data);
+
+  env.tracker.record({
+    message: "Publishing unknown block should fail",
+    slot: env.clock.currentSlot,
+    assertionId: "unknownBlockParent",
+  });
+} catch (error) {
+  if (!(error as Error).message.includes("BLOCK_ERROR_PARENT_UNKNOWN")) {
+    env.tracker.record({
+      message: `Publishing unknown block should return "BLOCK_ERROR_PARENT_UNKNOWN" got "${(error as Error).message}"`,
+      slot: env.clock.currentSlot,
+      assertionId: "unknownBlockParent",
+    });
+  }
+}
+await waitForNodeSync(unknownBlockSync.nodePair, env.options.controller.signal);
```

Comment:
This condition is not enough. You are just checking that at some point sync status it not syncing. You must assert that the head of this node is exactly `head`. Since this is not normal usage I can't do any guarantees of what will the syncing status do. Please **do not use syncing status in unknown root sync test**. For the rest of usage of `waitForNodeSync` you must assert that the node declares itself as synced + syncs past a specific head too.

Reply:
I've reviewed the logs on the unknown block sync node completes syncing at

```
Eph 6/6 0.986[unknown-block-sync-node-cl-lodestar/chain] [36mverbose[39m: New chain head headSlot=51, headRoot=0x74e92383d9084442240ee46fd1aaaeb3fed1bfd504b3764c48c700bc9e92e122, delaySec=12.986000061035156
Eph 6/6 0.986[unknown-block-sync-node-cl-lodestar/chain] [36mverbose[39m: Block processed slot=51, root=0x74e92383d9084442240ee46fd1aaaeb3fed1bfd504b3764c48c700bc9e92e122, delaySec=12.986000061035156
Eph 6/6 1.258[unknown-block-sync-node-cl-lodestar/rest] [34mdebug[39m: Req req-g 127.0.0.1 getSyncingStatus
Eph 6/6 1.259[unknown-block-sync-node-cl-lodestar/rest] [34mdebug[39m: Res req-g getSyncingStatus - 200
Eph 6/6 2.001[unknown-block-sync-node-cl-lodestar] [32minfo[39m: Synced - slot: 54 (skipped 3) - head: 51 0x74e9…e122 - execution: valid(0xc9a2…a9bb) - 
```

The initial error is triggered at
```
Eph 6/4 0.208[unknown-block-sync-node-cl-lodestar/chain] [31merror[39m: Block error slot=51 code=BLOCK_ERROR_PARENT_UNKNOWN, parentRoot=0x62bfece4644b0c9901e8298eda2c926800991418308b09bb15c7b9e7326ef8f9
```
Which is the second to last block that's processed
```
Eph 6/6 0.971[unknown-block-sync-node-cl-lodestar/chain] [36mverbose[39m: New chain head headSlot=50, headRoot=0x62bfece4644b0c9901e8298eda2c926800991418308b09bb15c7b9e7326ef8f9, delaySec=16.97099995613098
```
Then the node stalls.

So yes, do not use syncing status here. The node returns not syncing just because it so happens to be close to the head, but that's a "coincidence".